bansa/src/main/kotlin/com/brianegan/bansa/RxStore.kt

```diff
@@ -1,31 +1,24 @@
 package com.brianegan.bansa
 
 import rx.Observable
-import rx.Scheduler
 import rx.Subscriber
 import rx.Subscription
-import rx.schedulers.Schedulers
 import rx.subjects.PublishSubject
 import rx.subjects.SerializedSubject
 
 class RxStore<S, A>(
-        private val initialState: S,
-        private val initialReducer: (S, A) -> S,
-        private val scheduler: Scheduler = Schedulers.newThread()
+        override var state: S,
+        val reducer: (S, A) -> S
 ) : Store<S, A> {
-    private val dispatcher: SerializedSubject<A, A>
-    override var state: S = initialState
+
+    private val dispatcher = SerializedSubject<A, A>(PublishSubject.create<A>())
 
     override val stateChanges: Observable<S>
-    var reducer: (S, A) -> S
 
     init {
-        reducer = initialReducer
-        dispatcher = SerializedSubject<A, A>(PublishSubject.create<A>())
         stateChanges = dispatcher // When an action is dispatched
-                .observeOn(scheduler) // Run the scan on a given thread, by default a background thread
```

Comment:
No problem, this discussion if more interesting than the code itself :-)

You're right, it's a premature optimization, and at the current point I'm 90% sure it's a bad idea so it will be interesting to reopen the discussion when someone asks it for a real use case.